developer/src/kmc-keyboard/src/util/util.ts

```diff
@@ -54,3 +54,25 @@ export function allUsedKeyIdsInLayers(layersList : LKLayers[] | null): Set<strin
   }
   return s;
 }
+
+/**
+ * Determine modifier from layer info
+ * @param layer layer obj
+ * @returns modifier
+ */
+export function translateLayerAttrToModifier(layer: LDMLKeyboard.LKLayer) : number {
+  const {id, modifier} = layer;
+  if(id == 'base') {
+    return constants.keys_mod_none;
+  }
+  if (modifier) {
+    let mod = constants.keys_mod_none;
+    for (let str of modifier.split(' ')) {
+      const submod = constants.keys_mod_map.get(str);
+      mod |= submod;
```

Comment:
Can we assume that the input string contains only valid modifier strings?

Reply:
if not, validation would have failed.